### PR TITLE
Replace remained OTE to OTX

### PR DESCRIPTION
**Pull request by JihwanEom**
### Summary
- Replace remained OTE to OTX, no side effect is expected.
- **Changes in otx.algorithms.action.tasks.train.py is to resolve mypy issue, same code with #1956.** 

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/action/tasks/train.py [hunk 1]
```diff
@@ -157,14 +157,9 @@ def _train_model(self, dataset: DatasetEntity):
         self._model.train()
         # FIXME runner is built inside of train_model funciton, it is hard to change runner's type
         train_model(model=self._model, dataset=mm_train_dataset, cfg=training_config, validate=True)
-        checkpoint_file_path = glob(os.path.join(self._recipe_cfg.work_dir, "best*pth"))
+        checkpoint_file_path = glob(os.path.join(self._recipe_cfg.work_dir, "best*pth"))[0]
```
Comment (JihwanEom) [inline otx/algorithms/action/tasks/train.py:160]: Changes in otx.algorithms.action.tasks.train.py is to resolve mypy issue, same code with https://github.com/openvinotoolkit/training_extensions/pull/1956.

File: otx/algorithms/anomaly/configs/classification/padim/template.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ summary: This model is faster and in many cases more accurate, but it requires a
 application: ~
 
 # Algo backend.
-framework: OTEAnomalyClassification v0.1.0
+framework: OTXAnomalyClassification v0.1.0
```
Comment (harimkang) [inline otx/algorithms/anomaly/configs/classification/padim/template.yaml:11]: As far as I know, this shouldn't be changed for Geti yet. (maybe..?, all templates)

**Review by harimkang — CHANGES_REQUESTED** [review]

File: otx/algorithms/anomaly/configs/classification/padim/template.yaml (hunk 2 above)
Comment (JihwanEom) [reply otx/algorithms/anomaly/configs/classification/padim/template.yaml:11]: Thanks for noticing, @samet-akcay could you double check?

Comment (harimkang) [reply otx/algorithms/anomaly/configs/classification/padim/template.yaml:11]: @goodsong81

Comment (goodsong81) [reply otx/algorithms/anomaly/configs/classification/padim/template.yaml:11]: It seems OK as Geti Platform team has just (yesterday...) merged the framework images into one Docker image (thanks to `pip install .[full]`).

Comment (goodsong81) [reply otx/algorithms/anomaly/configs/classification/padim/template.yaml:11]: image

**Review by harimkang — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**